waiter/test/waiter/util/async_utils_test.clj

```diff
@@ -351,3 +351,34 @@
       (is (every? #(>= % (- interval-ms tolerance-ms)) invocation-diffs)
           (str {:invocation-diffs invocation-diffs
                 :invocation-times invocation-times})))))
+
+(deftest test-throttle-chan
+  (let [num-iterations 10
+        source-delay-ms 100
+        target-throttle-ms 400
+        source-chan (latest-chan)
+        target-chan (throttle-chan target-throttle-ms [source-chan])
+        target-data-atom (atom [])
+        process-complete-latch (CountDownLatch. 1)]
```

Comment:
You can use a promise chan and do `(async/<!! process-complete-promise)`

Reply:
Done.